scipy/stats/tests/test_mstats_basic.py

```diff
@@ -223,6 +231,13 @@ def test_kendalltau(self):
                             25,80,80,80,80,80,80, 0,10,45, np.nan, 0])
         result = mstats.kendalltau(x,y)
         assert_almost_equal(np.asarray(result), [-0.1585188, 0.4128009])
+        # make sure internal variable use correct precision with
+        # larger arrays
+        x = np.arange(2000, dtype=np.float)
+        x = ma.masked_greater(x, 1995)
+        y = np.arange(2000, dtype=np.float)
+        y = np.concatenate((y[1000:], y[:1000]))
+        assert_almost_equal(mstats.kendalltau(x,y)[1], 0.97344095)
```

Comment:
This number is not nearly as easy to compute "by hand" as the Spearman's rho above. The overflow bug was actually causing the p-value to be NaN, since the overflow caused a negative number to be passed to a square root along the way. We could replace this assertion with one that just checks for a real value, but as I said above that is not as robust against overflow errors none-the-less creating values that are at least reasonable (if the value over-flowed but created instead created a positive number going into the square root then the p-value would necessarily have been returned in the `[0, 1]` range).

I'm open to advice on what to do here, since as you say the value is not entirely reproducible. I might be able to spend some time trying to come up with how to calculate it by hand but that would have to wait until tonight (UTC-6).